pennylane/plugins/tests/conftest.py

```diff
@@ -0,0 +1,177 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Contains shared fixtures for the device tests."""
+import os
+import pytest
+import numpy as np
+import pennylane as qml
+
+# ==========================================================
+# pytest fixtures
+
+# seed for random functions
+np.random.seed(42)
+# Tolerance for analytic tests
+TOL = 1e-6
+# Tolerance for non-analytic tests
+TOL_STOCHASTIC = 0.05
+# Number of shots to call the devices with
+N_SHOTS = 10000
+# List of all devices that are included in PennyLane
+LIST_CORE_DEVICES = {"default.qubit", "default.qubit.tf"}

```

Comment:
Curios for the reason for leaving out `default.tensor`?

Reply:
I wasn't sure if the beta devices are strictly speaking core devices?

Reply:
I tried to add default.tensor and default.tensor.tf, and it turns out they do not have an "analytic" attribute. 

This is one of those issues with the PR - the tests are supposed to work for the existing devices, but at the same time they should define what a device must be able to do. I would vote for making "analytic" a compulsory attribute (or capability), since everything hinges on it and leaving the tests as is for now?

Reply:
Oh, good point. As far as I know, both devices are (by default) `analytic=True`, they don't support finite sampling. They should really define the attribute --- I agree with making it mandatory.

Reply:
The reason these devices don't have an `analytic` attribute is simply because they do not inherit from `QubitDevice`. Doesn't mean they aren't actually "analytic" though :) 

I feel this is something that is better declared in `_capabilities` though